MetaMask/metamask-mobile
# Fix/4754 importing wallet via qr

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

Issue reported by @olenapankina & @cortisiko where importing an account via QR scanning wasn't working when the QR scan is accessed through the account selector. This PR refactors the `ImportPrivateKey` component and fixes the originally found issue. Note - Import via QR code from the Wallet view is unaffected. That logic is separate.

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Test**
Since this is based off of the Permission System branch, there are two main cases to test:
1. Ensure that the original issue is fixed. You should be able to import an account via scanning QR code when accessing it through the account selector.
2. Ensure that importing via QR code while in the PS connect flow doesn't dismiss the account selector after you're done.
